libs/designsystem/helpers/src/ionic-element-part-helper.ts

```diff
@@ -7,7 +8,9 @@ export class IonicElementPartHelper {
   public async setPart(partName: string, ionicElementRef: ElementRef, selector: string) {
     // Ensure custom element has been defined and rendered
     await customElements.whenDefined(ionicElementRef.nativeElement.localName);
-    const customElement = await ionicElementRef.nativeElement.componentOnReady();
+    const customElement = await new Promise<HTMLElement>((resolve) => {
+      componentOnReady(ionicElementRef.nativeElement, (element) => resolve(element));
+    });
```

Comment:
I think we should still wrap it in a promise, so that the promise returned from the setPart method resolves after all the work is done.

We could create a util function that wraps the ionic function and returns a promise:
```ts
import { componentOnReady as ionicComponentOnReady } from '@ionic/core';

export async function componentOnReady(element: Element): Promise<Element> {
  return await new Promise<Element>((resolve) => {
    ionicComponentOnReady(element, resolve);
  });
}
```

And then use it like this:
```ts
  public async setPart(partName: string, ionicElementRef: ElementRef, selector: string) {
    // Ensure custom element has been defined and rendered
    await customElements.whenDefined(ionicElementRef.nativeElement.localName);
    const customElement = await componentOnReady(ionicElementRef.nativeElement);

    const partElement: HTMLElement = customElement.shadowRoot.querySelector(selector);
    if (!partElement) return;

    this.renderer.setAttribute(partElement, 'part', partName);
  }
```

Reply:
Good point. Everywhere we are using setPart() just executes it synchronosly, so the promise seems redundant 🤔 

I guess without async await it becomes a couple of nested callbacks, and my original purpose was to enhance readability... So it is fine by me to leave it as is.